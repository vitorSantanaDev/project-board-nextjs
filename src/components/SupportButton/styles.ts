import styled from 'styled-components'

export const Wrapper = styled.div`
  position: absolute;

  right: 50px;
  bottom: 70px;
  z-index: 999;
`

export const ButtonSupport = styled.button`
  width: 100px;
  height: 100px;

  font-size: 1.6rem;

  border: 0;
  border-radius: 50%;

  background-color: #ffb800;
  filter: drop-shadow(2px 1px 5px rgba(0, 0, 0, 0.3));

  cursor: pointer;
  transition: ease-in-out 0.3s;

  &:hover {
    transform: scale(1.05);
  }
`
