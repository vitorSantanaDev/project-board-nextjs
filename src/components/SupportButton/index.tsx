import Link from 'next/link'
import * as S from './styles'

const SupportButton: React.FC = () => {
  return (
    <S.Wrapper>
      <Link href="/donate">
        <S.ButtonSupport>Apoiar</S.ButtonSupport>
      </Link>
    </S.Wrapper>
  )
}

export default SupportButton
