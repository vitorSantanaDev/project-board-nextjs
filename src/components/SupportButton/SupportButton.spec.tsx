import { render } from '@testing-library/react'
import SupportButton from '.'

describe('<SupportButton />', () => {
  it('should render the heading', () => {
    render(<SupportButton />)
  })

  it('should match snapshot', () => {
    const { container } = render(<SupportButton />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
