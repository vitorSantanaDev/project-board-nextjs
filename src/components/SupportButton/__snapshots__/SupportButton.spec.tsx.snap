// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SupportButton /> should match snapshot 1`] = `
.c0 {
  position: absolute;
  right: 50px;
  bottom: 70px;
  z-index: 999;
}

.c1 {
  width: 100px;
  height: 100px;
  font-size: 1.6rem;
  border: 0;
  border-radius: 50%;
  background-color: #ffb800;
  -webkit-filter: drop-shadow(2px 1px 5px rgba(0,0,0,0.3));
  filter: drop-shadow(2px 1px 5px rgba(0,0,0,0.3));
  cursor: pointer;
  -webkit-transition: ease-in-out 0.3s;
  transition: ease-in-out 0.3s;
}

.c1:hover {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

<div
  class="c0"
>
  <button
    class="c1"
  >
    Apoiar
  </button>
</div>
`;
